### PR TITLE
Add `catalog` parameter to spec to `/index/summary` endpoints (#2000)

### PR DESCRIPTION
**Pull request by amarjandu**

### CODE REVIEW

**Comment by amarjandu** [comment]
> Looks great.
> I considered requesting you add another commit to fix the misspelled `seach` in the function name, however it's not really related to what you're doing here so maybe we'll save that for another day.

I did that in #1947, I think it got merged over the weekend. ill have to rebase.

**Comment by amarjandu** [comment]
TODO:
add `catalog` parameter required in `/index/projects`

File: lambdas/service/app.py [hunk 1]
```diff
@@ -869,15 +869,16 @@ def repository_search(entity_type: str, item_id: str):
     '''),
 )
 
+catalog_param_spec = params.query('catalog',
+                                  schema.optional(schema.with_default(config.catalog,
+                                                                      type_=schema.pattern(IndexName.catalog_name_re))),
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:874]: Line is way too long.

Comment (hannes-ucsc) [reply lambdas/service/app.py:874]:
```diff
Index: lambdas/service/app.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lambdas/service/app.py	(revision 663b7dab382d472836f85d8bef8f584516d9afbb)
+++ lambdas/service/app.py	(date 1595008515576)
@@ -869,10 +869,12 @@
     '''),
 )
 
-catalog_param_spec = params.query('catalog',
-                                  schema.optional(schema.with_default(config.catalog,
-                                                                      type_=schema.pattern(IndexName.catalog_name_re))),
-                                  description='The name of the catalog to query.')
+catalog_param_spec = params.query(
+    'catalog',
+    schema.optional(schema.with_default(config.catalog,
+                                        type_=schema.pattern(IndexName.catalog_name_re))),
+    description='The name of the catalog to query.'
+)
 
 
 def repository_search_params_spec(index_name):
```

File: lambdas/service/app.py [hunk 2]
```diff
@@ -1208,7 +1212,7 @@ def get_order():
             'token',
             schema.optional({'type': 'string'}),
             description='Reserved. Do not pass explicitly.',
-        ),
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:1211]: This causes a conflict now.

Comment (amarjandu) [reply lambdas/service/app.py:1211]: resolved merge conflicts after rebasing with `develop` branch.

**Pull request merged**